packages/sui-ssr/server/ssr/index.js

```diff
@@ -70,22 +70,35 @@ export default (req, res, next) => {
       const device = buildDeviceFrom({request: req})
       const [criticalHTML, bodyHTML] = html.split('</head>')
 
-      res.type('html')
-      res.write(criticalHTML)
-      res.flush()
-
       const context = await contextFactory(
         createServerContextFactoryParams(req)
       )
-      const {
-        initialProps,
-        reactString,
-        performance
-      } = await ssrComponentWithInitialProps({
-        context: {...context, device},
-        renderProps,
-        Target: withContext(context)(RouterContext)
-      })
+
+      let initialData
+
+      try {
+        initialData = await ssrComponentWithInitialProps({
+          context: {...context, device},
+          renderProps,
+          Target: withContext(context)(RouterContext)
+        })
+      } catch (err) {
+        // Managing 500 exception
+        return next(err)
+      }
+
+      const {initialProps, reactString, performance} = initialData
+
+      if (initialProps.error) {
```

Comment:
Maybe we could use a less common name to this. What do you think about use: `__error__` like key ?!

Reply:
Or even be more explicit and have to import something from sui-ssr like we did with the hooks.

Reply:
It's a common name because it is an Error instance indeed. I can't imagine a simpler and clearer contract. But I'm open to change it if you have a clear suggestion about the name. Regarding hooks, If this contract is clear and it's not prone to change, I don't really see the point of implementing it in some other way. Thoughts?

Reply:
`getInitialProps` already returns a Promise. Why do not use the `catch` in order to handle the error instead having to return an error property? Maybe some changes are needed, or not, in react-initial-props package but it sounds like it makes more sense.

Reply:
In fact, the error of the `getInitialProps` could be captured here: https://github.com/SUI-Components/sui/pull/462/files/3eb50222dea1bbc6eafb7ce165f336acd9170605#diff-c1aac530bdd240d83babeab7fefaf59aR86 and no `error` property would be needed.

Reply:
I am thinking that maybe this place is not the best place to do that. Because what we want is to avoid is render our page if we have an Error in the getInitialprops function. But the current GIP package will pass the error prop to our page component. Is that what we want ?!